packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_filter_dialog.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import '../../../../../shared/common_widgets.dart';
+import '../../../../../shared/dialogs.dart';
+import '../../../../../shared/theme.dart';
+import '../../../../../shared/utils.dart';
+import '../../../shared/heap/class_filter.dart';
+import '../controller/utils.dart';
+
+class ClassFilterDialog extends StatefulWidget {
+  const ClassFilterDialog(
+    this.classFilter, {
+    super.key,
+    required this.onChanged,
+  });
+
+  final ClassFilter classFilter;
+  final Function(ClassFilter filter) onChanged;
+
+  @override
+  State<ClassFilterDialog> createState() => _ClassFilterDialogState();
+}
+
+class _ClassFilterDialogState extends State<ClassFilterDialog> {
+  bool _initialized = false;
+  late String _rootPackage;
+
+  late ClassFilterType _type;
+  final _except = TextEditingController();
+  final _only = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_initialize());
+  }
+
+  Future<void> _initialize() async {
+    assert(!_initialized);
+    _rootPackage = await tryToDetectRootPackage() ?? '';
+    if (_rootPackage.isNotEmpty) _rootPackage = '$_rootPackage/';
+    _loadStateFromFilter(widget.classFilter);
+    setState(() => _initialized = true);
+  }
+
+  @override
+  void didUpdateWidget(covariant ClassFilterDialog oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.classFilter != widget.classFilter) {
+      _loadStateFromFilter(widget.classFilter);
+    }
+  }
+
+  void _loadStateFromFilter(ClassFilter filter) {
+    _type = filter.filterType;
+    _except.text = filter.except;
+    _only.text = filter.only ?? _rootPackage;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (!_initialized) return const CenteredCircularProgressIndicator();
+
+    final textFieldLeftPadding = scaleByFontFactor(40.0);
+    void onTypeChanged(ClassFilterType? type) => setState(() => _type = type!);
+
+    return StateUpdateDialog(
+      title: 'Filter Classes and Packages',
+      helpText: _helpText,
+      onResetDefaults: () =>
+          setState(() => _loadStateFromFilter(ClassFilter.empty())),
+      onApply: () {
+        final newFilter = ClassFilter(
+          filterType: _type,
+          except: _except.text,
+          only: _only.text,
+        );
+        widget.onChanged(newFilter);
+      },
+      onCancel: null,
```

Comment:
Can we make onCancel non-required instead of forcing an explicit passing of null? This is pretty standard for Flutter